src/main/java/com/amazon/opendistroforelasticsearch/security/transport/OpenDistroSecurityRequestHandler.java

```diff
@@ -126,16 +126,9 @@ protected void messageReceivedDecorate(final T request, final TransportRequestHa
 
             String channelType = transportChannel.getChannelType();
 
-            if(!channelType.equals("direct") && !channelType.equals("transport")) {
-                Class wrappedChannelCls = transportChannel.getClass();
-
-                try {
-                    Method getInnerChannel = wrappedChannelCls.getMethod("getInnerChannel", null);
-                    TransportChannel innerChannel = (TransportChannel)(getInnerChannel.invoke(transportChannel));
-                    channelType = innerChannel.getChannelType();
-                } catch (NoSuchMethodException ex) {
-                    throw new RuntimeException("Unknown channel type " + channelType + " does not implement getInnerChannel method.");
-                }
+            if (!channelType.equals("direct") && !channelType.equals("transport")) {
+                TransportChannel innerChannel = getInnerChannel(transportChannel);
+                channelType = innerChannel.getChannelType();
```

Comment:
I am not sure if we should channel to innerChannel here.

Reply:
Afaik its fine to change channel to innerChannel here. 

In our previous PR's we override the channelType value instead of the channel value but the idea behind the two seem to be the same - https://github.com/opendistro-for-elasticsearch/security-ssl/pull/2/files; https://github.com/opendistro-for-elasticsearch/security/pull/3/files.